bin/compile

```diff
@@ -1,77 +1,105 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
+
 function log() {
-        echo "-----> $*"
+    echo "-----> $*"
 }
 
-export BUILD_DIR=$1
-export ENV_DIR=$3
-export POETRY_VERSION="$(cat $ENV_DIR/POETRY_VERSION 2>/dev/null)"
+function indent() {
+    sed -e 's/^/       /'
+}
+
+export BUILD_DIR="$1"
+export ENV_DIR="$3"
+
+BUILDPACK_VARIABLES="DISABLE_POETRY_CREATE_RUNTIME_FILE PYTHON_RUNTIME_VERSION"
 
-export DISABLE_POETRY_CREATE_RUNTIME_FILE="$(cat $ENV_DIR/DISABLE_POETRY_CREATE_RUNTIME_FILE 2>/dev/null || true)"
+shopt -s nullglob
 
-if [ -z "$POETRY_VERSION" ];
-then
+for env_file in "$ENV_DIR"/POETRY_* ; do
+    export "$(basename "$env_file")=$(cat "$env_file" 2>/dev/null)"
+done
+
+for env_file in $BUILDPACK_VARIABLES ; do
+    if [ -f "$ENV_DIR/$env_file" ] ; then
+        export "$(basename "$env_file")=$(cat "$ENV_DIR/$env_file" 2>/dev/null)"
+    fi
+done
+
+if [ -z "${POETRY_VERSION:-}" ] ; then
     export POETRY_VERSION=1.1.4
     log "No Poetry version specified in POETRY_VERSION config var. Defaulting to $POETRY_VERSION."
 else
     log "Using Poetry version from POETRY_VERSION config var: $POETRY_VERSION"
 fi
 
-function indent() {
-        c='s/^/       /'
-        case $(uname) in
-                Darwin) sed -l "$c";;
-                *)      sed -u "$c";;
-        esac
-}
-
 log "Generate requirements.txt with Poetry"
 
 log "Install Poetry"
 
 curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py \
-  | sed -e 's/allowed_executables = \["python", "python3"\]/allowed_executables = ["python3"]/' \
-  | python3 \
-  | indent
+    | sed -e 's/allowed_executables = \["python", "python3"\]/allowed_executables = ["python3"]/' \
+    | python3 \
+    | indent
+
+PATH_TO_POETRY_ENV="$HOME/.poetry/env"
 
-source $HOME/.poetry/env
+if [ -f "$PATH_TO_POETRY_ENV" ] ; then
+    # shellcheck source=/dev/null
```

Comment:
debug remnant?

Reply:
Nope, this is a special directive to tell `shellcheck` that it should ignore dynamically sourced file during inspection.